stubs/docutils/docutils/transforms/__init__.pyi

```diff
@@ -1,14 +1,31 @@
-from _typeshed import Incomplete
+from collections.abc import Iterable
+from types import ModuleType
+from typing import Any
 
-from docutils.nodes import Node, document
+from docutils import ApplicationError, Component, TransformSpec, nodes
+from docutils.nodes import Node
+
+class TransformError(ApplicationError): ...
 
 class Transform:
```

Comment:
```suggestion
class Transform(ABC):
```

I'm not sure whether this should really be done here because the implementation does not explicitly inherit from ABC. However, `apply` raise `NotImplementedError`, so it *is* an abstract class per se. However, I'm not sure if (runtime) subclasses would possibly be marked as conflicting if they have a different metaclass (that is not compatible with `ABCMeta`, e.g., it does not directly inherit from it (at runtime, again)).

Reply:
@JelleZijlstra? what's the correct idiom here?

Reply:
We could use `@abstractmethod` without inheriting from ABC, so that at least mypy will complain if it isn't overridden in a subclass. This is documented in [these mypy docs](https://mypy.readthedocs.io/en/stable/class_basics.html#abstract-base-classes-and-multiple-inheritance).

Reply:
Oh I wasn't aware of that. Let's go for that then !